6_drivers/src/nccopy_utils.R

```diff
@@ -1,26 +1,125 @@
+  y <- parse_nc_filename(nc_filename, 'y')
+  var <- parse_nc_filename(nc_filename, 'var')
+  t0 <- seq(time_range[1], to = time_range[2], by = max_steps)
+  if(tail(t0, 1L) == time_range[2]){ # the somewhat rare case where the time start sequence includes the final step too
+    t0 <- head(t0, -1L)
+  }
+  t1 <- c(tail(t0, -1L) -1, time_range[2])
+
+  temp_nc_dir <- file.path(tempdir(), '.nc_split_files')
+  dir.create(temp_nc_dir)
+  # delete all temporary split files and their directory:
+  on.exit(unlink(temp_nc_dir, recursive = TRUE))
+
+  split_file_metadata <- sapply(1:length(t0), function(i) {
+    create_nc_filename(t0[i], t1[i], x0 = x[1], x1 = x[2], y0 = y[1], y1 = y[2], variable = var)
+  }, USE.NAMES = FALSE)
+
+  split_file_info <- data.frame(file_meta = split_file_metadata, file_num = 1:length(split_file_metadata)) %>%
+    mutate(nc_temp_file_unl = file.path(temp_nc_dir, sprintf("NLDAS_%s_unlimited.nc", stringr::str_pad(file_num, width = 3, pad = '0')))) %>%
+    mutate(nc_temp_file_fixed = file.path(temp_nc_dir, sprintf("NLDAS_%s_fixed.nc", stringr::str_pad(file_num, width = 3, pad = '0'))))
+
+  pb <- progress_bar$new(
+    format = "  nccopy :what [:bar] :percent eta: :eta",
+    clear = TRUE, total = length(t0))
+
+  for (split_file in split_file_metadata){
+
+    nc_temp_file_fixed <- split_file_info %>% filter(file_meta == split_file) %>% pull(nc_temp_file_fixed)
+    nc_temp_file_unl <- split_file_info %>% filter(file_meta == split_file) %>% pull(nc_temp_file_unl)
+
+    nldas_url <- nldas_url_from_file(split_file)
+    nccopy_retry(nldas_url, nc_temp_file_fixed)
+
+    # make the time dimension unlimited so we can append to this file:
+    system(sprintf("ncks --mk_rec_dmn time %s -o %s", nc_temp_file_fixed, nc_temp_file_unl))
+
+    what_download <- strsplit(nldas_url, split = '[?]')[[1]][2]
+    pb$tick(tokens = list(what = what_download))
+  }
+
+  split_filepaths <- paste(pull(split_file_info, nc_temp_file_unl), collapse = ' ')
+
+  combine_nc_files(split_filepaths, nc_filepath)
+
+  # check for timestep order here??
+
+  # check that the resulting file has the appropriate number of timesteps:
+  verify_nc_output(nc_filepath)
+
+  invisible(nc_filepath)
+}
+
+
+#' take a bunch of nc files that share the same variable and combine them into a single file
+#' needs to remove time as a record dimension w/ ncks
+combine_nc_files <- function(filepaths_to_combine, nc_out_filepath){
+
+  nc_unl_filepath <- file.path(tempdir(), "NLDAS_COMBINED_unl.nc")
+  # need NCO to run this. Combine all split files into one:
+  system(sprintf('ncrcat -h --overwrite %s %s', filepaths_to_combine, nc_unl_filepath))
+
+
+  if (file.exists(nc_out_filepath)){
+    unlink(nc_out_filepath) # doesn't seem to honor the overwrite arg and gets hung up on interactive mode
+  }
+
+  # then make the time dim fixed again on the larger file
+  system(sprintf("ncks --fix_rec_dmn time %s -o %s", nc_unl_filepath, nc_out_filepath))
```

Comment:
Not sure why the overwrite flag wouldn't work.  What were you trying?

Reply:
I tried `--overwrite` and some other ones specified in the docs. Almost seemed like a bug or something...

Reply:
hmm, maybe there was an issue with the arg order?  It happens in the terminal too, not just from `system()` in R?

Reply:
Could report it on sourceforge then: https://sourceforge.net/p/nco/discussion/9829.

`nco` is up to date?

Reply:
I don't think I did an exhaustive test before moving on w/ the simple `unlink` fix.